models/postgreSql_model_queries.js

```diff
@@ -0,0 +1,144 @@
+//NOTE: for performance, prefix all terminal commands with:
+EXPLAIN ANALYZE
+
+//POSTGRES Syntax for Read (GET) using itemId
+//ATTEMPT 1 as two different requests
+SELECT
+  *
+FROM
+   items
+WHERE
+  item_Id = itemId; //is it single ='s for equal to in postgres?
+
+//would have to pass value of items.item_StoreId to next query
+SELECT
+  *
+FROM
+  stores
+WHERE
+  store_id = //item_storeid passed in from first query;  //does a query have access to both tables as long as you're in the same db?
+
+//ATTEMPT 2 as 1 request using JOIN
+SELECT * FROM items
+JOIN stores ON items.item_StoreId = stores.store_id
+WHERE items.item_Id = /*insert id or id's here */;
+
+SELECT items.item_price, stores.store_name, stores.store_address, stores.store_phonenumber FROM items
+JOIN stores ON items.item_StoreId = stores.store_id
+WHERE items.item_Id = 10000000;
+
+
+//GET - find one item
+let findAnItemAvailAndStore = function (itemId) {
+  console.log('db model rcvd item id = ', itemId)
+  return ItemAvailability.findOne({ itemId: itemId }, '-_id -__v')
+    .populate({
+      path: 'itemAvailability',
+      populate: {
+        path: 'storeId'
+      }
+    })
+    .then((data) => {
+      console.log('db model processed itemId, store lookup = ', data);
+      if (data) {
+        let storeData = data.itemAvailability.map((store) => {
+          return {
+            storeName: store.storeId.storeName,
+            storeAddress: store.storeId.storeAddress,
+            storePhoneNumber: store.storeId.storePhoneNumber,
+            availability: store.availability
+          }
+        })
+        return { itemAvailability: storeData };
+      } else {
+        return { itemAvailability: 'No items found'};
+      }
+    })
+    .catch((err) => {
+      res.status(500).send(err);
+      console.log(err);
+    })
+};
```

Comment:
i'm interested: if no items are found does the db controller throw an error instead of running the else statement in the then block?

Reply:
Another great question and point. Will test this and get back to you.